# Analyza liekov podla skupiny ATC7 
* pozn. Doplnit info o liekoch, popis!!, upravit grafy a farby

Analýzu sme robili na dátach z roku 2019, keďže je to posledný rok, z ktorého sme mali údaje za všetky štvrťroky a zároveň v tomto roku na Slovensku ešte nebola pandémia.
Otázky:
* Ktoré lieky s danou účinnou látkou sa predávajú viac než ostatné? Závisí predaj od krajiny výrobcu?

(mozno fajn vo vsetkych grafoch krajiny rovnakou farbou)
 1409 roznych ucinnych latok

In [9]:
import pandas as pd 
import numpy as np 
import matplotlib.pyplot as plt 
import plotly.express as px
%run tables.py

# Tabulka, s ktorou pracujeme (rok 2019) s vybranymi udajmi

In [10]:
table = import_table('detailed', 2019)
table = table.rename(columns={'Počet balení': 'Počet_balení'})

In [11]:

table2019=table.loc[:,['ATC7_KOD','ATC7_popis','LIEK_KOD','LIEK_popis', 'STAT_popis', 'VYROBCA_popis', 'Počet_balení','Úhrada_PAC']]
table2019 = table2019.query('ATC7_KOD != "(prazdne)"').query( 'ATC7_popis != "neznámy"')
table2019sum = table2019.groupby(['ATC7_KOD','ATC7_popis','LIEK_KOD','LIEK_popis', 'STAT_popis','VYROBCA_popis']).sum().reset_index()
display(table2019sum.head())
display(table2019sum.tail())


ATC7_KOD                           ATC7_popis LIEK_KOD  \
0  A01AA03                              Olaflur   C40630   
1  A01AA30  Profylaxia zubného kazu, kombinácie   C40631   
2  A01AA30  Profylaxia zubného kazu, kombinácie   C85867   
3  A01AB03                         Chlorhexidín   C14106   
4  A01AB03                         Chlorhexidín   C34296   

                         LIEK_popis STAT_popis  \
0       ELMEX FLUID sol dnt 1x50 ml    Nemecko   
1        ELMEX gelee gel dnt 1x25 g    Nemecko   
2       ELMEX gelee gel dnt 1x215 g    Nemecko   
3         CORSODYL sol ora 1x600 ml      Česko   
4  PROTEFIX ANTISEPT sol der 200 ml    Nemecko   

                                       VYROBCA_popis  Počet_balení  Úhrada_PAC  
0                                          GABA GmbH          40.0      135.08  
1                                       CP GABA GmbH       43832.5   313077.14  
2                                       CP GABA GmbH          29.0      171.85  
3  GlaxoSmithKline Consumer Healthcare Czech Repu...          72.0      254.33  
4                         Queisser Pharma GmbH & Co.           6.0       27.82

ATC7_KOD   ATC7_popis LIEK_KOD                        LIEK_popis  \
9813      V12  Homeopatiká   C82842  GINKGO BILOBA plv 15 g ch2-ch200   
9814      V12  Homeopatiká   C96701  ARNICA MONTANA ung 20 g ch2-ch30   
9815      V12  Homeopatiká   C97944                   QUIETUDE tbl 60   
9816      V12  Homeopatiká   C99176       BELLADONNA gra 4 g ch3-ch30   
9817      V12  Homeopatiká   C99177   ARNICA MONTANA gra 4 g ch2-ch30   

      STAT_popis        VYROBCA_popis  Počet_balení  Úhrada_PAC  
9813  Francúzsko  Laboratoires BOIRON         309.0      882.18  
9814  Francúzsko  Laboratoires BOIRON          15.0       52.29  
9815  Francúzsko  Laboratoires BOIRON         156.0     1030.21  
9816  Francúzsko  Laboratoires BOIRON         183.0      631.73  
9817  Francúzsko  Laboratoires BOIRON         700.0     2375.06

# Tabulka: 15 najpredavanejsich ucinnych latok

In [12]:
atc7 = table2019.loc[:,['ATC7_KOD', 'ATC7_popis','Počet_balení']].groupby(['ATC7_KOD', 'ATC7_popis']).sum().sort_values(by='Počet_balení', ascending=False).reset_index().head(15)
display(atc7)

ATC7_KOD                                  ATC7_popis  Počet_balení
0   V09FX01                              Technécium - (   9618479.510
1   B05BB01                                 Elektrolyty   5331686.852
2   N02BE01                                 Paracetamol   3297020.107
3   M01AE01                                   Ibuprofén   2795946.717
4   C07AB07                                  Bisoprolol   2607372.542
5   C08CA01                                   Amlodipín   2119340.883
6   N05BA12                                  Alprazolam   2045121.730
7   M01AB05                                  Diklofenak   2013726.518
8   N02BB02                        Metamizol, sodná soľ   1946250.707
9   C07AB02                                  Metoprolol   1873602.519
10  B01AC06                   Kyselina acetylsalicylová   1751434.007
11  C10AA05                                Atorvastatín   1570661.378
12  N05CF02                                    Zolpidém   1555453.927
13  R01AA07                               Xylometazolín   1496827.500
14  R02AA20                 Rôzne iné krčné antiseptiká   1444293.959
15  N02BE51  Paracetamol, kombinácie okrem psycholeptík   1322071.122
16  M02AA15                                  Diklofenak   1185154.441
17  A11GA01              Kyselina askorbová (vitamín C)   1167817.511
18  A02BC02                                 Pantoprazol   1151639.605
19  R01AA09                                  Tramazolín   1135312.849

In [37]:
fig = px.bar(atc7.sort_values(by='Počet_balení'), x='Počet_balení', y='ATC7_popis', orientation='h')
fig.show()

## Ukazky udajov jednotlivych tabuliek (podla poctu baleni)

In [13]:
nazvy = []
for n in atc7['ATC7_KOD']:
    if n not in nazvy:
        nazvy.append(n)

top20tab= {}
for i in range(20):
    top20tab[nazvy[i]] = table2019sum.query(f'ATC7_KOD == "{nazvy[i]}"').sort_values(by='Počet_balení', ascending= False).reset_index()
    display(top20tab[nazvy[i]].head())

index ATC7_KOD      ATC7_popis LIEK_KOD  \
0   9547  V09FX01  Technécium - (   C91102   
1   9546  V09FX01  Technécium - (   C15457   
2   9541  V09FX01  Technécium - (   C15308   
3   9545  V09FX01  Technécium - (   C15456   
4   9540  V09FX01  Technécium - (   C15307   

                                          LIEK_popis STAT_popis  \
0                             Drytec gen rnk 2,5 GBq  Holandsko   
1                            Drytec gen rnk 10,0 GBq  Holandsko   
2  Ultra-TechneKow FM (99mTC-generátor) inj 10,63...  Holandsko   
3                             Drytec gen rnk 9,0 GBq  Holandsko   
4  Ultra-TechneKow FM (99mTC-generátor) inj 8,50 ...  Holandsko   

             VYROBCA_popis  Počet_balení  Úhrada_PAC  
0       GE Healthcare B.V.    6688913.50         0.0  
1       GE Healthcare B.V.    1469870.00         0.0  
2  Curium Netherlands B.V.     991979.00         0.0  
3       GE Healthcare B.V.     467689.01         0.0  
4  Curium Netherlands B.V.         14.00         0.0

index ATC7_KOD   ATC7_popis LIEK_KOD  \
0   1450  B05BB01  Elektrolyty   C37743   
1   1432  B05BB01  Elektrolyty   C33593   
2   1431  B05BB01  Elektrolyty   C33590   
3   1505  B05BB01  Elektrolyty   C97748   
4   1506  B05BB01  Elektrolyty   C97749   

                                          LIEK_popis STAT_popis  \
0  0,9% SODIUM CHL.IN W.F.I.FRESENIUS sol inf 1x1...      Česko   
1  0,9% SODIUM ChL.IN W.F.I.FRESENIUS sol inf 1x2...      Česko   
2  0,9% SODIUM ChL.IN W.F.I.FRESENIUS sol inf 1x5...      Česko   
3  SODIUM CHLORIDE 0,9%W/V I.V. INF inf 1x100 ml ...  Taliansko   
4  SODIUM CHLORIDE 0,9%W/V I.V. INF inf 1x250 ml ...  Taliansko   

           VYROBCA_popis  Počet_balení  Úhrada_PAC  
0  Fresenius Kabi s.r.o.    1115466.27     1758.64  
1  Fresenius Kabi s.r.o.     999336.36     4893.52  
2  Fresenius Kabi s.r.o.     789546.85     1375.02  
3  Bieffe Medital S.p.A.     418659.15     5287.02  
4  Bieffe Medital S.p.A.     381477.64    18777.30

index ATC7_KOD   ATC7_popis LIEK_KOD  \
0   6299  N02BE01  Paracetamol   C86052   
1   6298  N02BE01  Paracetamol   C86051   
2   6287  N02BE01  Paracetamol   C27613   
3   6290  N02BE01  Paracetamol   C57476   
4   6281  N02BE01  Paracetamol   C26627   

                                          LIEK_popis STAT_popis  \
0            PARALEN 500 tbl 24x500 mg (blis.Al/PVC)  Slovensko   
1            PARALEN 500 tbl 12x500 mg (blis.Al/PVC)  Slovensko   
2                           PARALEN 100 sup 5x100 mg  Slovensko   
3  Panadol Paracetamolum 500 mg tbl flm 24x500 mg...      Česko   
4                          PARALEN 125 tbl 20x125 mg  Slovensko   

                                       VYROBCA_popis  Počet_balení  Úhrada_PAC  
0                     sanofi-aventis Slovakia s.r.o.   2300126.084  3826254.68  
1                     sanofi-aventis Slovakia s.r.o.    269140.730   279950.53  
2                     sanofi-aventis Slovakia s.r.o.    139481.200   117364.89  
3  GlaxoSmithKline Consumer Healthcare Czech Repu...    113876.250   194443.83  
4                     sanofi-aventis Slovakia s.r.o.    111906.328    81286.36

index ATC7_KOD ATC7_popis LIEK_KOD  \
0   5604  M01AE01  Ibuprofén   C35458   
1   5640  M01AE01  Ibuprofén   C7880A   
2   5651  M01AE01  Ibuprofén   C97299   
3   5618  M01AE01  Ibuprofén   C49851   
4   5610  M01AE01  Ibuprofén   C45451   

                                          LIEK_popis  STAT_popis  \
0                      IBALGIN 400 tbl flm 24x400 mg   Slovensko   
1       Ibalgin 400 tbl flm 48x400 mg (blis. PVC/Al)   Slovensko   
2                       BRUFEN 400 tbl flm 30x400 mg  Francúzsko   
3                          MIG-400 tbl flm 30x400 mg     Nemecko   
4  NUROFEN pre deti Jahoda sus por 1x100 ml (PET ...       Česko   

                                       VYROBCA_popis  Počet_balení  Úhrada_PAC  
0                     sanofi-aventis Slovakia s.r.o.    596621.970  1885611.99  
1                     sanofi-aventis Slovakia s.r.o.    293384.731  1432962.47  
2                             Mylan Laboratories SAS    232798.061   868313.80  
3                                   Berlin-Chemie AG    205849.323   706016.17  
4  Reckitt Benckiser (Czech Republic), spol. s r.o.     160812.710   747388.09

index ATC7_KOD  ATC7_popis LIEK_KOD  \
0   2001  C07AB07  Bisoprolol   C94164   
1   1997  C07AB07  Bisoprolol   C90028   
2   1974  C07AB07  Bisoprolol   C33446   
3   1971  C07AB07  Bisoprolol   C33427   
4   2004  C07AB07  Bisoprolol   C97761   

                                          LIEK_popis STAT_popis  \
0                           CONCOR 5 tbl flm 30x5 mg  Slovensko   
1  CORONAL 5 tbl flm 30x5 mg (PVC/PE/PVDC/Alu blis.)  Slovensko   
2             BISOPROLOL-ratiopharm 5 mg tbl 30x5 mg    Nemecko   
3                CONCOR COR 2,5 mg tbl flm 30x2,5 mg  Slovensko   
4                       BISOMERCK  5 tbl flm 30x5 mg  Slovensko   

        VYROBCA_popis  Počet_balení  Úhrada_PAC  
0  Merck spol. s.r.o.     861847.12  2312824.20  
1           ZNT-SK-2?     439825.33   198924.15  
2     ratiopharm GmbH     361429.00    71700.98  
3  Merck spol. s.r.o.     174083.00   439473.10  
4  Merck spol. s.r.o.     116380.50   156601.14

index ATC7_KOD ATC7_popis LIEK_KOD  \
0   2084  C08CA01  Amlodipín   C13556   
1   2094  C08CA01  Amlodipín   C37500   
2   2089  C08CA01  Amlodipín   C34886   
3   2105  C08CA01  Amlodipín   C62569   
4   2085  C08CA01  Amlodipín   C13558   

                                          LIEK_popis STAT_popis  \
0                                 AGEN 5 tbl 30x5 mg      Česko   
1  Amlodipin Stichting 5 mg tablety tbl 30x5 mg (...  Slovinsko   
2              AMLODIPIN-RATIOPHARM 5 mg tbl 30x5 mg    Nemecko   
3                                NORVASC tbl 30x5 mg   Belgicko   
4                               AGEN 10 tbl 30x10 mg      Česko   

                 VYROBCA_popis  Počet_balení  Úhrada_PAC  
0                Zentiva, k.s.    890258.100   201748.72  
1  Sandoz Pharmaceuticals d.d.    359816.500   104900.67  
2              ratiopharm GmbH    320902.970    78281.35  
3        Pfizer Europe MA EEIG    101640.000   341657.38  
4                Zentiva, k.s.     94816.326   105940.59

index ATC7_KOD  ATC7_popis LIEK_KOD                          LIEK_popis  \
0   7195  N05BA12  Alprazolam   C33494  NEUROL 0,5 mg tbl 30x0,5 mg (bli.)   
1   7203  N05BA12  Alprazolam   C91788   NEUROL 0,25 tbl 30x0,25 mg (bli.)   
2   7189  N05BA12  Alprazolam   C14596         FRONTIN 0,5MG tbl 30x0,5 mg   
3   7199  N05BA12  Alprazolam   C86656    NEUROL 1,0 tbl 30x1,00 mg (bli.)   
4   7187  N05BA12  Alprazolam   C14594       FRONTIN 0,25MG tbl 30x0,25 mg   

  STAT_popis             VYROBCA_popis  Počet_balení  Úhrada_PAC  
0      Česko             Zentiva, k.s.      478721.0   625761.97  
1      Česko             Zentiva, k.s.      360820.0   351477.76  
2   Maďarsko  EGIS Pharmaceuticals PLC      256590.2   346183.19  
3      Česko             Zentiva, k.s.      216768.5   418523.86  
4   Maďarsko  EGIS Pharmaceuticals PLC      190090.0   180653.82

index ATC7_KOD  ATC7_popis LIEK_KOD  \
0   5492  M01AB05  Diklofenak   C55701   
1   5476  M01AB05  Diklofenak   C23849   
2   5510  M01AB05  Diklofenak   C89136   
3   5501  M01AB05  Diklofenak   C67547   
4   5480  M01AB05  Diklofenak   C27802   

                                          LIEK_popis STAT_popis  \
0                 FLECTOR EP RAPID 50MG gra 20x50 mg  Slovensko   
1  Voltaren Rapid 25 mg cps mol 20x25 mg (blis.PV...      Česko   
2    DICLOFENAC DUO PHARMAVIT 75 mg cps mod 30x75 mg      Česko   
3                          ALMIRAL inj 10x3 ml/75 mg     Cyprus   
4              VOLTAREN RAPID 50 mg tbl obd 20x50 mg  Slovensko   

                                       VYROBCA_popis  Počet_balení  Úhrada_PAC  
0                               IBSA Slovakia s.r.o.     755843.97  2614771.88  
1  GlaxoSmithKline Consumer Healthcare Czech Repu...     184241.68  1082657.96  
2                 PharmaSwiss Česká republika s.r.o.     170025.01   469116.41  
3                                    Medochemie Ltd.     167764.83   104939.22  
4                           Novartis Slovakia s.r.o.     163277.90   145603.20

index ATC7_KOD            ATC7_popis LIEK_KOD  \
0   6248  N02BB02  Metamizol, sodná soľ   C24857   
1   6245  N02BB02  Metamizol, sodná soľ   C24459   
2   6257  N02BB02  Metamizol, sodná soľ   C7520A   
3   6255  N02BB02  Metamizol, sodná soľ   C6590B   
4   6246  N02BB02  Metamizol, sodná soľ   C24530   

                                          LIEK_popis STAT_popis  \
0                  NOVALGIN 500 mg tbl flm 20x500 mg  Slovensko   
1       NOVALGIN 500 mg/1 ml sol inj 10x2 ml/1000 mg  Slovensko   
2  Metamizol Stada 500 mg/ml perorálne roztokové ...    Nemecko   
3     METAMISTAD tablety tbl 50x500 mg (blis.PVC/Al)    Nemecko   
4        NOVALGIN 500 mg/1 ml sol inj 5x5 ml/2500 mg  Slovensko   

                    VYROBCA_popis  Počet_balení  Úhrada_PAC  
0  sanofi-aventis Slovakia s.r.o.   1342217.310  1574143.21  
1  sanofi-aventis Slovakia s.r.o.    122225.700   119052.25  
2           STADA Arzneimittel AG     97835.600    90901.77  
3           STADA Arzneimittel AG     96200.357    53449.98  
4  sanofi-aventis Slovakia s.r.o.     96195.000    43043.01

index ATC7_KOD  ATC7_popis LIEK_KOD  \
0   1945  C07AB02  Metoprolol   C28641   
1   1937  C07AB02  Metoprolol   C14959   
2   1941  C07AB02  Metoprolol   C25634   
3   1944  C07AB02  Metoprolol   C28639   
4   1954  C07AB02  Metoprolol   C69827   

                                          LIEK_popis STAT_popis  \
0         BETALOC ZOK 50MG (blist.) tbl mod 28x50 mg  Taliansko   
1        BETALOC ZOK 25 mg (blist.) tbl mod 28x25 mg  Taliansko   
2          VASOCARDIN 50 tbl 50(5x10)x50 mg (blist.)  Slovensko   
3      BETALOC ZOK 100 mg (blist.) tbl mod 28x100 mg  Taliansko   
4  Metoprolol ratiopharm 50 mg tbl plg 30x50 mg (...    Nemecko   

                                       VYROBCA_popis  Počet_balení  Úhrada_PAC  
0  Recordati Industria Chimica e Farmaceutica S.p.A.      415989.0  1335363.74  
1  Recordati Industria Chimica e Farmaceutica S.p.A.      301260.5  1055435.89  
2                                          ZNT-SK-2?      253627.8    11447.21  
3  Recordati Industria Chimica e Farmaceutica S.p.A.      194362.0   600822.26  
4                                    ratiopharm GmbH      168777.1    28732.82

index ATC7_KOD                 ATC7_popis LIEK_KOD  \
0   1000  B01AC06  Kyselina acetylsalicylová   C23029   
1   1001  B01AC06  Kyselina acetylsalicylová   C23030   
2    996  B01AC06  Kyselina acetylsalicylová   C12460   
3   1002  B01AC06  Kyselina acetylsalicylová   C3492A   
4   1008  B01AC06  Kyselina acetylsalicylová   C5131A   

                                          LIEK_popis STAT_popis  \
0        ANOPYRIN 100 mg tbl 56x100 mg (blis.PVC/Al)  Slovensko   
1        ANOPYRIN 100 mg tbl 84x100 mg (blis.PVC/Al)  Slovensko   
2              ASPIRIN PROTECT 100 tbl ent 50x100 mg  Slovensko   
3  STADAPYRIN 100 mg tbl ent 60x100 mg (blis.PVC/Al)    Nemecko   
4  STADAPYRIN 100 mg tbl ent 120x100 mg (blis.PVC...    Nemecko   

           VYROBCA_popis  Počet_balení  Úhrada_PAC  
0              ZNT-SK-2?    537364.368   939250.30  
1              ZNT-SK-2?    419356.344  1009800.82  
2    Bayer, spol. s.r.o.    241531.880   783576.97  
3  STADA Arzneimittel AG    105627.101   207325.82  
4  STADA Arzneimittel AG     95864.974   346927.37

index ATC7_KOD    ATC7_popis LIEK_KOD  \
0   2847  C10AA05  Atorvastatín   C3053B   
1   2880  C10AA05  Atorvastatín   C40541   
2   2879  C10AA05  Atorvastatín   C40538   
3   2863  C10AA05  Atorvastatín   C38841   
4   2864  C10AA05  Atorvastatín   C38846   

                                          LIEK_popis STAT_popis  \
0  Torvacard Novum 20 mg tbl flm 90x20 mg (blis.O...      Česko   
1                         ATORIS 20 tbl flm 90x20 mg  Slovinsko   
2                         ATORIS 20 tbl flm 30x20 mg  Slovinsko   
3  Atorvastatin Actavis 20 mg tbl flm 30x20 mg (b...     Island   
4  Atorvastatin Actavis 20 mg tbl flm 100x20 mg (...     Island   

           VYROBCA_popis  Počet_balení  Úhrada_PAC  
0          Zentiva, k.s.    137486.552   297993.59  
1  KRKA d.d., Novo mesto    133908.820   382533.66  
2  KRKA d.d., Novo mesto    111426.090   114511.80  
3  Actavis Group PTC ehf     94341.990    31342.27  
4  Actavis Group PTC ehf     91083.300    74300.28

index ATC7_KOD ATC7_popis LIEK_KOD                             LIEK_popis  \
0   7250  N05CF02   Zolpidém   C68902                  STILNOX 10 mg TBL FLM   
1   7255  N05CF02   Zolpidém   C83373                  STILNOX 10 mg TBL FLM   
2   7247  N05CF02   Zolpidém   C46861              HYPNOGEN tbl flm 15x10 mg   
3   7245  N05CF02   Zolpidém   C14610          SANVAL 10 mg tbl flm 20x10 mg   
4   7256  N05CF02   Zolpidém   C92955  ZOLPIDEM MERCK 10 mg tbl flm 20x10 mg   

  STAT_popis                   VYROBCA_popis  Počet_balení  Úhrada_PAC  
0  Slovensko  sanofi-aventis Slovakia s.r.o.     442430.66  1012808.95  
1  Slovensko  sanofi-aventis Slovakia s.r.o.     423611.00   437661.24  
2    Nemecko                     Takeda GmbH     248027.35   394571.60  
3  Slovinsko     Sandoz Pharmaceuticals d.d.     177540.10   184548.25  
4      ĺrsko           Mylan Ireland Limited      97213.24   154214.22

index ATC7_KOD     ATC7_popis LIEK_KOD  \
0   7812  R01AA07  Xylometazolín   C76962   
1   7800  R01AA07  Xylometazolín   C36097   
2   7798  R01AA07  Xylometazolín   C2872A   
3   7807  R01AA07  Xylometazolín   C68978   
4   7799  R01AA07  Xylometazolín   C36096   

                                          LIEK_popis STAT_popis  \
0                    OLYNTH 0,1 % aer nao 10 ml 0,1%      ĺrsko   
1                    OLYNTH HA 0,1 % aer nao 1x10 ml      ĺrsko   
2  Otrivin Menthol 0,1% aer nao 1x10 ml (fľ.HDPE ...      Česko   
3        mar rhino 0,1% nosový sprej aer nao 1x15 ml    Nemecko   
4                   OLYNTH HA 0,05 % aer nao 1x10 ml      ĺrsko   

                                       VYROBCA_popis  Počet_balení  Úhrada_PAC  
0                McNeil Healthcare (Ireland) Limited      553939.0  2079755.78  
1                McNeil Healthcare (Ireland) Limited      456407.0  2190324.28  
2  GlaxoSmithKline Consumer Healthcare Czech Repu...      133135.0   681134.20  
3                              STADA Arzneimittel AG       90071.0   339016.91  
4                McNeil Healthcare (Ireland) Limited       72639.0   350843.34

index ATC7_KOD                   ATC7_popis LIEK_KOD  \
0   7900  R02AA20  Rôzne iné krčné antiseptiká   C26747   
1   7925  R02AA20  Rôzne iné krčné antiseptiká   C62547   
2   7932  R02AA20  Rôzne iné krčné antiseptiká   C67008   
3   7936  R02AA20  Rôzne iné krčné antiseptiká   C96521   
4   7911  R02AA20  Rôzne iné krčné antiseptiká   C44219   

                          LIEK_popis STAT_popis  \
0             Dorithricin pas ord 20    Nemecko   
1  STREPSILS Med a Citrón pas ord 24      Česko   
2     NEO-ANGIN bez cukru pas ord 24    Nemecko   
3               SEPTISAN tbl 20x5 mg  Slovensko   
4        STREPSILS PLUS pas ora 1x24      Česko   

                                       VYROBCA_popis  Počet_balení  Úhrada_PAC  
0                        Medice Pharma GmbH & Co. KG    653423.650  4647661.34  
1  Reckitt Benckiser (Czech Republic), spol. s r.o.     149267.740   944644.61  
2                                    Divapharma GmbH     94564.240   612499.19  
3                     sanofi-aventis Slovakia s.r.o.     90881.000   301647.58  
4  Reckitt Benckiser (Czech Republic), spol. s r.o.      76438.189   564480.76

index ATC7_KOD                                  ATC7_popis LIEK_KOD  \
0   6367  N02BE51  Paracetamol, kombinácie okrem psycholeptík   C65951   
1   6334  N02BE51  Paracetamol, kombinácie okrem psycholeptík   C30229   
2   6342  N02BE51  Paracetamol, kombinácie okrem psycholeptík   C33813   
3   6347  N02BE51  Paracetamol, kombinácie okrem psycholeptík   C36516   
4   6311  N02BE51  Paracetamol, kombinácie okrem psycholeptík   C0381A   

                                          LIEK_popis STAT_popis  \
0  TheraFlu prechladnutie a chrípka plo por 1x14 ...      Česko   
1                            PARALEN PLUS tbl flm 24  Slovensko   
2  THERAFLU PRECHLADNUTIE A CHRÍPKA plo por 1x10 ...      Česko   
3             PARALEN HOT DRINK plu por 1x12 vreciek  Slovensko   
4  PARALEN GRIP chrípka a bolesť tbl flm 1x24 (bl...  Slovensko   

                                       VYROBCA_popis  Počet_balení  Úhrada_PAC  
0  GlaxoSmithKline Consumer Healthcare Czech Repu...    422718.171  3954421.86  
1                     sanofi-aventis Slovakia s.r.o.    100459.030   589486.22  
2  GlaxoSmithKline Consumer Healthcare Czech Repu...     81452.991   671984.16  
3                     sanofi-aventis Slovakia s.r.o.     65515.562   481851.44  
4                     sanofi-aventis Slovakia s.r.o.     61588.890   362517.93

index ATC7_KOD  ATC7_popis LIEK_KOD  \
0   5780  M02AA15  Diklofenak   C11849   
1   5781  M02AA15  Diklofenak   C11851   
2   5785  M02AA15  Diklofenak   C18515   
3   5784  M02AA15  Diklofenak   C18514   
4   5808  M02AA15  Diklofenak   C84453   

                                          LIEK_popis STAT_popis  \
0  Voltaren Forte 2,32 % gél gel der 1x100 g (tub...      Česko   
1  Voltaren Forte 2,32 % gél gel der 1x150 g (tub...      Česko   
2                      FLECTOR EP TISSUGEL emp tdm 5  Slovensko   
3                             FLECTOR EP gel 1x100 g  Slovensko   
4                              DICLOBENE gel 1x100 g  Holandsko   

                                       VYROBCA_popis  Počet_balení  Úhrada_PAC  
0  GlaxoSmithKline Consumer Healthcare Czech Repu...     189743.17  2256703.40  
1  GlaxoSmithKline Consumer Healthcare Czech Repu...     136412.90  2083185.45  
2                               IBSA Slovakia s.r.o.     136175.20  1201954.19  
3                               IBSA Slovakia s.r.o.     112500.25   884399.45  
4                                          TEV-NL-4?      94815.70   515849.66

index ATC7_KOD                      ATC7_popis LIEK_KOD  \
0    782  A11GA01  Kyselina askorbová (vitamín C)   C78278   
1    781  A11GA01  Kyselina askorbová (vitamín C)   C78277   
2    778  A11GA01  Kyselina askorbová (vitamín C)   C42420   
3    777  A11GA01  Kyselina askorbová (vitamín C)   C35913   
4    771  A11GA01  Kyselina askorbová (vitamín C)   C24836   

                                          LIEK_popis STAT_popis  \
0                        CELASKON 250 tbl 100x250 mg  Slovensko   
1                         CELASKON 250 tbl 30x250 mg  Slovensko   
2  CELASKON 500 mg červený pomaranč tbl eff 30x50...  Slovensko   
3             CELASKON LONG EFFECT cps pld 60x500 mg  Slovensko   
4    ACIDUM ASCORBICUM BIOTIKA sol inj 5x5 ml/500 mg      Česko   

                    VYROBCA_popis  Počet_balení  Úhrada_PAC  
0  sanofi-aventis Slovakia s.r.o.    396438.278  1796114.28  
1  sanofi-aventis Slovakia s.r.o.    250880.500   389292.20  
2  sanofi-aventis Slovakia s.r.o.    106336.777   602810.11  
3  sanofi-aventis Slovakia s.r.o.     79906.524   654347.40  
4                  BB Pharma a.s.     79579.400    42800.62

index ATC7_KOD   ATC7_popis LIEK_KOD  \
0    138  A02BC02  Pantoprazol   C43345   
1    139  A02BC02  Pantoprazol   C43349   
2    142  A02BC02  Pantoprazol   C43362   
3    141  A02BC02  Pantoprazol   C43358   
4    144  A02BC02  Pantoprazol   C43791   

                                          LIEK_popis STAT_popis  \
0              Nolpaza 20 mg tbl ent 30x20 mg (bli.)  Slovinsko   
1             Nolpaza 20 mg tbl ent 100x20 mg (bli.)  Slovinsko   
2             Nolpaza 40 mg tbl ent 100x40 mg (bli.)  Slovinsko   
3              Nolpaza 40 mg tbl ent 30x40 mg (bli.)  Slovinsko   
4  Pantoprazol Sandoz 20 mg gastrorezistentné tab...  Slovinsko   

                 VYROBCA_popis  Počet_balení  Úhrada_PAC  
0        KRKA d.d., Novo mesto    184810.510   255770.78  
1        KRKA d.d., Novo mesto    144399.293   632243.24  
2        KRKA d.d., Novo mesto    117848.280   648739.25  
3        KRKA d.d., Novo mesto    107996.080   297846.23  
4  Sandoz Pharmaceuticals d.d.     73014.340   105977.50

index ATC7_KOD  ATC7_popis LIEK_KOD                      LIEK_popis  \
0   7819  R01AA09  Tramazolín   C26797  MUCONASAL PLUS sol neb 1x10 ml   

  STAT_popis                   VYROBCA_popis  Počet_balení  Úhrada_PAC  
0  Slovensko  sanofi-aventis Slovakia s.r.o.   1135312.849  5548815.77

# Grafy pre jednotlive ucinne latky

# 1. Technecium

In [14]:
technecium = top20tab["V09FX01"]

for i,popis in enumerate(technecium['LIEK_popis']):
    if 'Drytec gen' in popis:
        technecium.iloc[i,4] = 'Drytec gen rnk'
    elif "Ultra-TechneKow" in popis:
        technecium.iloc[i,4] = 'Ultra-TechneKow FM (99mTC-generátor)'

technecium = technecium.groupby(['LIEK_popis', 'STAT_popis', 'VYROBCA_popis']).sum().reset_index()
display(technecium)
print()
print("Typ vydaju:")
item_counts = table.query('ATC7_KOD == "V09FX01"')["TYP_VYDAJ_POP"].value_counts(normalize=True)
print(item_counts)



LIEK_popis STAT_popis            VYROBCA_popis  \
0                        Drytec gen rnk  Holandsko       GE Healthcare B.V.   
1  Ultra-TechneKow FM (99mTC-generátor)  Holandsko  Curium Netherlands B.V.   

   index  Počet_balení  Úhrada_PAC  
0  57267    8626479.51         0.0  
1  38158     992000.00         0.0


Typ vydaju:
výdaj z nemocničnej lekárne pre vlastnú nemocnicu    0.692308
úhrada z verejnej poisťovne                          0.307692
Name: TYP_VYDAJ_POP, dtype: float64


In [15]:
fig = px.bar(technecium, x='LIEK_popis', y='Počet_balení', color='VYROBCA_popis', title='Technecium')
fig.show()

 Na vizualizaciu ucinnej latky technecium nam bolo najvyhodnejsie pouzit barplot. 
 Nakolko sa jedna o diagnosticke radiofarmaka, podla vypisu Series vidime ze tieto lieky boli poskytovane nemocnicou alebo ich
 pacientom hradila zdravotna poistovna. V oboch pripadoch pacient neplatil nic, a tak udaje o cene nemame ako zakomponovat. Obe radiofarmaka su vyrabane Holandskymi vyrobcami, preto v grafe nie su rozlisene na zaklade krajiny ale vyrobcu.

# 2. Elektrolyty:

In [56]:
elektrolyty = top20tab["B05BB01"]

fig = px.bar(elektrolyty.groupby(['VYROBCA_popis','STAT_popis']).sum().reset_index().query('Počet_balení > 500000'), x='VYROBCA_popis', y='Počet_balení',title='Elektrolyty podla vyrobcov nad 500 000', color='STAT_popis')
fig.show()

In [58]:
elektrolyty_staty = elektrolyty.groupby('STAT_popis').sum().reset_index()
fig = px.pie(elektrolyty_staty, names='STAT_popis', values='Počet_balení', title='Elektrolyty podla krajiny vyrobcu (vsetky)')
fig.show()

Vidime, ze najviac (61.8%) pouzivame ceskych elektorlytov, co je vyse 3 000 000. Slovenske elektrolyty su az na poslednom mieste iba so 4.83% co je priblizne 250 000.

# 3. Paracetamol (30 najpredavanejsich)

In [17]:
paracetamol = top20tab["N02BE01"].sort_values(by='Počet_balení', ascending=False).head(30)

for i,popis in enumerate(paracetamol['LIEK_popis']):
    if 'MEDIPYRIN 500 tbl' in popis:
        paracetamol.iloc[i,4] = 'MEDIPYRIN tbl'
    elif 'PANADOL BABY (+odmerka)' in popis:
        paracetamol.iloc[i,4] = 'PANADOL BABY (+odmerka)'
    elif 'PANADOL BABY (+striekačka)' in popis:
        paracetamol.iloc[i,4] = 'PANADOL BABY (+striekačka)'
    elif 'PANADOL BABY' in popis:
        paracetamol.iloc[i,4] = 'PANADOL BABY'
    elif 'PANADOL JUNIOR' in popis:
        paracetamol.iloc[i,4] = 'PANADOL JUNIOR'
    elif 'PANADOL PRE DETI' in popis:
        paracetamol.iloc[i,4] = 'PANADOL PRE DETI'
    elif 'PANADOL tbl' in popis:
        paracetamol.iloc[i,4] = 'PANADOL tbl'
    elif 'PARALEN' in popis and 'sus' in popis:
        paracetamol.iloc[i,4] = 'PARALEN sus'
    elif 'PARALEN' in popis and 'tbl' in popis:
        paracetamol.iloc[i,4] = 'PARALEN tbl'
    elif 'PARALEN' in popis and 'sup' in popis:
        paracetamol.iloc[i,4] = 'PARALEN sup'
    elif 'PARALEN' in popis:
        paracetamol.iloc[i,4] = 'PARALEN nápoj'
    elif 'Panadol Novum' in popis:
        paracetamol.iloc[i,4] = 'Panadol Novum tbl'
    elif 'Panadol Paracetamolum' in popis:
        paracetamol.iloc[i,4] = 'Panadol Paracetamolum tbl'
    elif 'Paracetamol B. Braun' in popis:
        paracetamol.iloc[i,4] = 'Paracetamol B. Braun'
    elif 'Paracetamol Kabi' in popis:
        paracetamol.iloc[i,4] = 'Paracetamol Kabi'
    elif 'Paracetamol Kappler' in popis:
        paracetamol.iloc[i,4] = 'Paracetamol Kappler'
    elif 'Paracetamol Vitabalans' in popis:
        paracetamol.iloc[i,4] = 'Paracetamol Vitabalans'

paracetamol = paracetamol.groupby(['LIEK_popis', 'STAT_popis', 'VYROBCA_popis']).sum().reset_index()
print("Typ vydaju:")
item_counts = table.query('ATC7_KOD == "N02BE01"')["TYP_VYDAJ_POP"].value_counts(normalize=True)
print(item_counts)

display(paracetamol.sort_values(by='LIEK_popis'))


Typ vydaju:
predané z verejnej lekárne bez receptu občanom (voľnopredajné lieky)    0.304054
výdaj z verejnej lekárne  na recept bez úhrady z verejného poistenia    0.209459
výdaj z verejnej lekárne na faktúru do neštátnych ambulancií            0.162162
výdaj z nemocničnej lekárne pre vlastnú nemocnicu                       0.162162
výdaj z verejnej lekárne na žiadanku nemocniciam                        0.141892
výdaj z nemocničnej lekárne pre zmluvnú nemocnicu                       0.013514
úhrada z verejnej poisťovne                                             0.006757
Name: TYP_VYDAJ_POP, dtype: float64


LIEK_popis  STAT_popis  \
0                MEDIPYRIN tbl       Česko   
1                 PANADOL BABY       Česko   
2      PANADOL BABY (+odmerka)       Česko   
3   PANADOL BABY (+striekačka)  Francúzsko   
4               PANADOL JUNIOR       Česko   
5             PANADOL PRE DETI       Česko   
6                  PANADOL tbl       Česko   
7                PARALEN nápoj   Slovensko   
8                  PARALEN sup   Slovensko   
9                  PARALEN sus   Slovensko   
10                 PARALEN tbl   Slovensko   
11           Panadol Novum tbl       Česko   
12   Panadol Paracetamolum tbl       Česko   
13        Paracetamol B. Braun     Nemecko   
14            Paracetamol Kabi       Česko   
15         Paracetamol Kappler     Nemecko   
16      Paracetamol Vitabalans      Fínsko   

                                        VYROBCA_popis  index  Počet_balení  \
0                     Glenmark Pharmaceuticals s.r.o.  18807     21920.890   
1   GlaxoSmithKline Consumer Healthcare Czech Repu...   6283     50394.000   
2   GlaxoSmithKline Consumer Healthcare Czech Repu...   6305       839.000   
3               SmithKline Beecham Liquides Industrie   6300       285.000   
4   GlaxoSmithKline Consumer Healthcare Czech Repu...   6285     30460.500   
5   GlaxoSmithKline Consumer Healthcare Czech Repu...  12579     37895.000   
6   GlaxoSmithKline Consumer Healthcare Czech Repu...  12580       553.000   
7                      sanofi-aventis Slovakia s.r.o.   6271       482.170   
8                      sanofi-aventis Slovakia s.r.o.  12565    161406.210   
9                      sanofi-aventis Slovakia s.r.o.   6292     74950.000   
10                     sanofi-aventis Slovakia s.r.o.  25139   2683311.142   
11  GlaxoSmithKline Consumer Healthcare Czech Repu...  12592     77893.815   
12  GlaxoSmithKline Consumer Healthcare Czech Repu...  12579    114351.250   
13                               B.Braun Melsungen AG  12519       900.500   
14                              Fresenius Kabi s.r.o.  12548     34926.290   
15                              STADA Arzneimittel AG   6288       663.330   
16                                      Vitabalans Oy  12529      5493.010   

    Úhrada_PAC  
0     36758.59  
1    100522.52  
2      2862.69  
3      1012.17  
4     77105.53  
5    140531.82  
6       942.77  
7      1673.49  
8    153352.77  
9    270530.14  
10  4190200.04  
11   159034.24  
12   195206.01  
13       55.58  
14    15860.05  
15     3072.16  
16      633.44

In [60]:
paracetamol_all = top20tab["N02BE01"].sort_values(by='Počet_balení', ascending=False)
for i,popis in enumerate(paracetamol_all['LIEK_popis']):
    if 'PANADOL' in popis:
        paracetamol_all.iloc[i,4] = 'PANADOL'
    elif 'Panadol' in popis:
        paracetamol_all.iloc[i,4] = 'PANADOL'
    elif 'PARALEN' in popis:
        paracetamol_all.iloc[i,4] = 'PARALEN'
    elif 'Paralen' in popis:
        paracetamol_all.iloc[i,4] = 'PARALEN'
    else:
        paracetamol_all.iloc[i,4] = 'INÉ'
paracetamol_all = paracetamol_all.groupby(['LIEK_popis']).sum().reset_index()
display(paracetamol_all)
fig = px.pie(paracetamol_all, values='Počet_balení', names='LIEK_popis', title='PARACETAMOL (vsetky)')
fig.show()

LIEK_popis   index  Počet_balení  Úhrada_PAC
0        INÉ   94046     63990.020    56460.27
1    PANADOL  163554    312880.565   677698.50
2    PARALEN   50267   2920149.522  4615756.44

Preco je paralen takto "popularny"? Monzno ze ho ludia viac poznaju, je to prvy liek s paracetamolom co ich napadne, mozno niektorym pride ucinnejsi, ale aj ked sa pozrieme na tabulku liekov s paracetamolom, vidime, ze panadol je zvycajne pre deti.

In [65]:
fig = px.bar(paracetamol,x='LIEK_popis', y='Počet_balení',color='STAT_popis', title='Paracetamol - pocet baleni (top 30)')
fig.update_xaxes(tickangle=45)
fig.show()

Z grafu vidime, ze vsetky lieky prekonava Paralen v tabletach. Tkisto si mozeme vsimnut ze je vyrabany na Slovensku. Cesko ma siroku ponuku Panadolov avsak tieto tak popularne nie su. Panadoly su zvycajne vo variantoch pre deti.

# 4. Ibuprofen

In [20]:
ibuprofen = top20tab["M01AE01"].sort_values(by='Počet_balení', ascending=False).head(30)

for i,popis in enumerate(ibuprofen['LIEK_popis']):
    if 'IBALGIN 400' in popis:
        ibuprofen.iloc[i,4] = 'IBALGIN 400'
    elif 'Ibalgin 400' in popis:
        ibuprofen.iloc[i,4] = 'IBALGIN 400'
    elif 'BRUFEN 400' in popis:
        ibuprofen.iloc[i,4] = 'BRUFEN 400'
    elif 'Brufen 400' in popis:
        ibuprofen.iloc[i,4] = 'BRUFEN 400'
    elif 'BRUFEN sirup' in popis:
        ibuprofen.iloc[i,4] = 'BRUFEN sirup'
    elif 'BRUFEN retard 800' in popis:
        ibuprofen.iloc[i,4] = 'BRUFEN retard 800'
    elif 'BRUFEN 600' in popis:
        ibuprofen.iloc[i,4] = 'BRUFEN 600'
    elif 'MIG-400' in popis:
        ibuprofen.iloc[i,4] = 'MIG-400'
    elif 'NUROFEN pre deti' in popis:
        ibuprofen.iloc[i,4] = 'NUROFEN pre deti'
    elif 'NUROFEN PRE DETI' in popis:
        ibuprofen.iloc[i,4] = 'NUROFEN pre deti'
    elif 'Nurofen pre deti' in popis:
        ibuprofen.iloc[i,4] = 'NUROFEN pre deti'
    elif 'NUROFEN PRE DETI' in popis:
        ibuprofen.iloc[i,4] = 'NUROFEN pre deti'
    elif 'IBALGIN RAPIDCAPS' in popis:
        ibuprofen.iloc[i,4] = 'IBALGIN RAPIDCAPS'
    elif 'IBALGIN 200' in popis:
        ibuprofen.iloc[i,4] = 'IBALGIN 200'
    elif 'IBALGIN 600' in popis:
        ibuprofen.iloc[i,4] = 'IBALGIN 600'
    elif 'IBALGIN FAST' in popis:
        ibuprofen.iloc[i,4] = 'IBALGIN FAST'
    elif 'NUROFEN Liquid Capsules' in popis:
        ibuprofen.iloc[i,4] = 'NUROFEN Liquid Capsules'
    elif 'IBUPROFEN 400' in popis:
        ibuprofen.iloc[i,4] = 'IBUPROFEN 400'
    elif 'IBALGIN SUS' in popis:
        ibuprofen.iloc[i,4] = 'IBALGIN SUS'
    elif 'MIG' in popis:
        ibuprofen.iloc[i,4] = 'MIG Junior'
    elif 'NUROFEN FORTE' in popis:
        ibuprofen.iloc[i,4] = 'NUROFEN FORTE'
    elif 'NUROFEN tbl' in popis:
        ibuprofen.iloc[i,4] = 'NUROFEN tbl'

ibuprofen = ibuprofen.groupby(['LIEK_popis', 'STAT_popis', 'VYROBCA_popis']).sum().reset_index()
display(ibuprofen)

LIEK_popis  STAT_popis  \
0                BRUFEN 400  Francúzsko   
1                BRUFEN 600  Francúzsko   
2               IBALGIN 200   Slovensko   
3               IBALGIN 400   Slovensko   
4               IBALGIN 600   Slovensko   
5              IBALGIN FAST   Slovensko   
6         IBALGIN RAPIDCAPS   Slovensko   
7               IBALGIN SUS   Slovensko   
8             IBUPROFEN 400     Nemecko   
9                MIG Junior     Nemecko   
10                  MIG-400     Nemecko   
11            NUROFEN FORTE       Česko   
12  NUROFEN Liquid Capsules       Česko   
13         NUROFEN pre deti       Česko   
14              NUROFEN tbl       Česko   

                                        VYROBCA_popis  index  Počet_balení  \
0                              Mylan Laboratories SAS  11226    260398.011   
1                              Mylan Laboratories SAS   5625     28425.100   
2                      sanofi-aventis Slovakia s.r.o.   5603    115937.580   
3                      sanofi-aventis Slovakia s.r.o.  22455    925557.576   
4                      sanofi-aventis Slovakia s.r.o.   5629     36402.670   
5                      sanofi-aventis Slovakia s.r.o.   5641     32353.100   
6                      sanofi-aventis Slovakia s.r.o.  11222    195327.956   
7                      sanofi-aventis Slovakia s.r.o.   5634     16038.000   
8                               STADA Arzneimittel AG  11253     87787.140   
9                                    Berlin-Chemie AG   5571     12276.300   
10                                   Berlin-Chemie AG  11235    234871.823   
11  Reckitt Benckiser (Czech Republic), spol. s r.o.    5627     53810.500   
12  Reckitt Benckiser (Czech Republic), spol. s r.o.   11245    140448.500   
13  Reckitt Benckiser (Czech Republic), spol. s r.o.   44911    518837.687   
14  Reckitt Benckiser (Czech Republic), spol. s r.o.    5633     53473.410   

    Úhrada_PAC  
0   1000652.93  
1     88020.41  
2    274563.27  
3   3503655.19  
4     64870.46  
5    108560.61  
6    899915.02  
7     52677.24  
8    228675.29  
9     34815.97  
10   758498.46  
11   186448.24  
12   557590.23  
13  2579581.43  
14   123508.07

In [66]:
fig = px.bar(ibuprofen, x='LIEK_popis', y='Počet_balení', color='STAT_popis', title='Ibuprofen - pocet baleni (top 30)')
fig.update_xaxes(tickangle=45)
fig.show()

In [70]:
ibuprofen_all = top20tab["M01AE01"].sort_values(by='Počet_balení', ascending=False)

for i,popis in enumerate(ibuprofen_all['LIEK_popis']):
    if 'IBALGIN' in popis:
        ibuprofen_all.iloc[i,4] = 'IBALGIN'
    elif 'Ibalgin' in popis:
        ibuprofen_all.iloc[i,4] = 'IBALGIN'
    elif 'BRUFEN' in popis:
        ibuprofen_all.iloc[i,4] = 'BRUFEN'
    elif 'Brufen' in popis:
        ibuprofen_all.iloc[i,4] = 'BRUFEN'
    elif 'NUROFEN' in popis:
        ibuprofen_all.iloc[i,4] = 'NUROFEN'
    elif 'Nurofen' in popis:
        ibuprofen_all.iloc[i,4] = 'NUROFEN'
    elif 'MIG' in popis:
        ibuprofen_all.iloc[i,4] = 'MIG'
    else:
        ibuprofen_all.iloc[i,4] = 'INÉ'
ibuprofen_all = ibuprofen_all.groupby(['LIEK_popis','STAT_popis']).sum().reset_index()
display(ibuprofen_all)
fig = px.pie(ibuprofen_all, values='Počet_balení', names='LIEK_popis', title='IBUPROFEN')
fig.show()
# Medzi ine su zaradene lieky s poctom baleni pod 100 000

LIEK_popis          STAT_popis   index  Počet_balení  Úhrada_PAC
0      BRUFEN          Francúzsko   67173    312661.631  1192883.73
1     IBALGIN           Slovensko  106811   1324939.472  4917048.03
2         INÉ              Fínsko   55935     16234.534    35680.04
3         INÉ           Holandsko    5586         5.000        0.20
4         INÉ             Nemecko   44889    100127.660   272368.37
5         INÉ             Rakúsko    5638       288.000      699.84
6         INÉ  Spojené kráľovstvo   16719        35.000      156.93
7         INÉ               ĺrsko   16908       896.900     4148.80
8         MIG             Nemecko   33625    247223.123   793578.27
9     NUROFEN          Francúzsko    5652         6.000       30.35
10    NUROFEN  Spojené kráľovstvo   16800       507.000     1581.02
11    NUROFEN               Česko  128939    793022.397  3558584.44

In [71]:
fig = px.bar(ibuprofen_all, y='Počet_balení', x='LIEK_popis', title='IBUPROFEN', color='STAT_popis')
fig.show()

Z grafu vidime, ze najpredavanejsi liek s ibuprofenom je slovensky Ibalgin 400. Z ceskych liekov bol najpredavanejsi Nurofen pre deti. To, ze bol druhy najpredavanejsi je mozno sposobene tym ze je prave pre deti. 

# 5. Bisoprolol

In [23]:
bisoprolol = top20tab["C07AB07"].sort_values(by='Počet_balení', ascending=False)

for i,popis in enumerate(bisoprolol['LIEK_popis']):
    if 'BISOBELA' in popis:
        bisoprolol.iloc[i,4] = 'BISOBELA'
    elif 'BISOCARD' in popis:
        bisoprolol.iloc[i,4] = 'BISOCARD'
    elif 'BISOGAMMA' in popis:
        bisoprolol.iloc[i,4] = 'BISOGAMMA'
    elif 'BISOMERCK' in popis:
        bisoprolol.iloc[i,4] = 'BISOMERCK'
    elif 'BISOPROLOL-ratiopharm' in popis:
        bisoprolol.iloc[i,4] = 'BISOPROLOL-ratiopharm'
    elif 'BIVAXOL' in popis:
        bisoprolol.iloc[i,4] = 'BIVAXOL'
    elif 'Bisoprolol Pfizer' in popis:
        bisoprolol.iloc[i,4] = 'Bisoprolol Pfizer'
    elif 'Bisoprolol Sandoz' in popis:
        bisoprolol.iloc[i,4] = 'Bisoprolol Sandoz'
    elif 'Bisprotin ' in popis:
        bisoprolol.iloc[i,4] = 'Bisprotin '
    elif 'CONCOR COR' in popis or 'Concor COR' in popis:
        bisoprolol.iloc[i,4] = 'CONCOR COR'
    elif 'CONCOR' in popis:
        bisoprolol.iloc[i,4] = 'CONCOR'
    elif 'CORONAL' in popis:
        bisoprolol.iloc[i,4] = 'CORONAL'
    elif 'RIVOCOR' in popis:
        bisoprolol.iloc[i,4] = 'RIVOCOR'
    elif 'Sobycor' in popis:
        bisoprolol.iloc[i,4] = 'Sobycor'

bisoprolol = bisoprolol.groupby(['LIEK_popis', 'STAT_popis', 'VYROBCA_popis']).sum().reset_index()
display(bisoprolol)

LIEK_popis          STAT_popis  \
0                BISOBELA           Slovensko   
1                BISOCARD               Česko   
2               BISOGAMMA             Nemecko   
3               BISOMERCK           Slovensko   
4   BISOPROLOL-ratiopharm             Nemecko   
5                 BIVAXOL               Česko   
6       Bisoprolol Pfizer  Spojené kráľovstvo   
7       Bisoprolol Sandoz           Slovinsko   
8              Bisprotin               Fínsko   
9                  CONCOR           Slovensko   
10             CONCOR COR           Slovensko   
11                CORONAL           Slovensko   
12                RIVOCOR               Česko   
13                Sobycor           Slovinsko   

                         VYROBCA_popis  index  Počet_balení  Úhrada_PAC  
0                       BELUPO, s.r.o.   5957      7197.330    16626.62  
1   PharmaSwiss Česká republika s.r.o.   7954    192394.600   151460.50  
2          Wörwag Pharma GmbH & Co.KG    1979         2.000        6.34  
3                   Merck spol. s.r.o.   4009    122739.500   174809.80  
4                      ratiopharm GmbH   3949    390962.830    86444.66  
5          IVAX Pharmaceuticals s.r.o.   1999         2.000        3.20  
6                Pfizer Europe MA EEIG   3933       119.000       16.40  
7          Sandoz Pharmaceuticals d.d.   3939     10493.660     2229.01  
8                        Vitabalans Oy   9920      1965.900     1922.87  
9                   Merck spol. s.r.o.   8006    920947.070  2677920.45  
10                  Merck spol. s.r.o.   7884    284506.250   747989.61  
11                           ZNT-SK-2?   5971    472688.870   237788.87  
12               PRO.MED.CS Praha a.s.   3955     98766.500    89467.99  
13               KRKA d.d., Novo mesto   9965    104587.032    82156.65

In [24]:
fig = px.bar(bisoprolol.query('Počet_balení > 90000'), x='LIEK_popis', y='Počet_balení', color='STAT_popis', title='Bisoprolol - balenia (nad 90 000)')
fig.update_xaxes(tickangle=45)
fig.show()

In [72]:
bisoprolol_krajina = bisoprolol.groupby('STAT_popis').sum().reset_index()
for i,popis in enumerate(bisoprolol_krajina['STAT_popis']):
    if 'Fínsko' in popis:
        bisoprolol_krajina.iloc[i,0] = 'iné'
    elif 'Spojené kráľovstvo' in popis:
        bisoprolol_krajina.iloc[i,0] = 'iné'
bisoprolol_krajina = bisoprolol_krajina.groupby('STAT_popis').sum().reset_index()
fig = px.pie(bisoprolol_krajina, values='Počet_balení', names='STAT_popis', title='BISOPROLOL')
fig.show()

Z grafu vidime, ze najpouzivanejsi betablokator s ucinnou latkou bisoprolol je slovensky Concor. Narozdiel od predchadzajucich grafov, vidime, ze aj ine lieky sa vo vyraznej miere pouzivaju, napr Coronal alebo nemecky Bisprolol-ratiopharm. Jednoznacne vedu slovenske lieky.

# 6. Amlodipin

In [26]:

amlodipin = top20tab["C08CA01"].sort_values(by='Počet_balení', ascending=False)

for i,popis in enumerate(amlodipin['LIEK_popis']):
    if 'ACCEL' in popis:
        amlodipin.iloc[i,4] = 'ACCEL'
    elif 'AGEN' in popis:
        amlodipin.iloc[i,4] = 'AGEN'
    elif 'AMLODIPIN HEXAL' in popis:
        amlodipin.iloc[i,4] = 'AMLODIPIN HEXAL'
    elif 'AMLODIPIN-RATIOPHARM' in popis:
        amlodipin.iloc[i,4] = 'AMLODIPIN-RATIOPHARM'
    elif 'AMLOPIN' in popis:
        amlodipin.iloc[i,4] = 'AMLOPIN'
    elif 'Amlodigamma' in popis:
        amlodipin.iloc[i,4] = 'Amlodigamma'
    elif 'Amlodipin Actavis' in popis:
        amlodipin.iloc[i,4] = 'Amlodipin Actavis'
    elif 'Amlodipin Stichting' in popis:
        amlodipin.iloc[i,4] = 'Amlodipin Stichting'
    elif 'Amlodipin Vitabalans' in popis:
        amlodipin.iloc[i,4] = '	Amlodipin Vitabalans'
    elif 'CARDILOPIN' in popis:
        amlodipin.iloc[i,4] = 'CARDILOPIN'
    elif 'NORVASC' in popis:
        amlodipin.iloc[i,4] = 'NORVASC'
    elif 'TENOX' in popis:
        amlodipin.iloc[i,4] = 'TENOX'

amlodipin = amlodipin.groupby(['LIEK_popis', 'STAT_popis', 'VYROBCA_popis']).sum().reset_index()
display(amlodipin)

LIEK_popis STAT_popis                 VYROBCA_popis  index  \
0   \tAmlodipin Vitabalans     Fínsko                 Vitabalans Oy   8326   
1                    ACCEL     Cyprus               Medochemie Ltd.   4199   
2                     AGEN      Česko                 Zentiva, k.s.   8352   
3          AMLODIPIN HEXAL     Dánsko                     HEXAL A/S   2106   
4     AMLODIPIN-RATIOPHARM    Nemecko               ratiopharm GmbH   4179   
5                  AMLOPIN  Slovinsko      LEK Pharmaceuticals d.d.   4205   
6              Amlodigamma    Nemecko  Artesan Pharma GmbH & Co. KG   2093   
7        Amlodipin Actavis     Island         Actavis Group PTC ehf   4187   
8      Amlodipin Stichting  Slovinsko   Sandoz Pharmaceuticals d.d.   6285   
9               CARDILOPIN   Maďarsko      EGIS Pharmaceuticals PLC   4175   
10                 NORVASC   Belgicko         Pfizer Europe MA EEIG   4209   
11                   TENOX  Slovinsko         KRKA d.d., Novo mesto   4195   

    Počet_balení  Úhrada_PAC  Cena úhrada_PAC/bal  
0       8133.700     8013.23             0.985189  
1      39635.000    36396.58             0.918294  
2    1042447.713   514840.56             0.493877  
3          2.000       13.46             6.730000  
4     402849.470    79580.14             0.197543  
5         13.000       16.80             1.292308  
6          2.000       30.12            15.060000  
7         16.000       13.50             0.843750  
8     415811.500   133785.38             0.321745  
9      62800.500    79712.18             1.269292  
10    123821.000   441976.19             3.569477  
11     23809.000    42794.71             1.797417

In [74]:

fig = px.bar(amlodipin.query('Počet_balení > 50000'), x='LIEK_popis', y='Počet_balení', color='STAT_popis', title='Amlodipin nad 50 000')
fig.update_xaxes(tickangle=45)
fig.show()
display(amlodipin)

LIEK_popis STAT_popis                 VYROBCA_popis  index  \
0   \tAmlodipin Vitabalans     Fínsko                 Vitabalans Oy   8326   
1                    ACCEL     Cyprus               Medochemie Ltd.   4199   
2                     AGEN      Česko                 Zentiva, k.s.   8352   
3          AMLODIPIN HEXAL     Dánsko                     HEXAL A/S   2106   
4     AMLODIPIN-RATIOPHARM    Nemecko               ratiopharm GmbH   4179   
5                  AMLOPIN  Slovinsko      LEK Pharmaceuticals d.d.   4205   
6              Amlodigamma    Nemecko  Artesan Pharma GmbH & Co. KG   2093   
7        Amlodipin Actavis     Island         Actavis Group PTC ehf   4187   
8      Amlodipin Stichting  Slovinsko   Sandoz Pharmaceuticals d.d.   6285   
9               CARDILOPIN   Maďarsko      EGIS Pharmaceuticals PLC   4175   
10                 NORVASC   Belgicko         Pfizer Europe MA EEIG   4209   
11                   TENOX  Slovinsko         KRKA d.d., Novo mesto   4195   

    Počet_balení  Úhrada_PAC  Cena úhrada_PAC/bal  
0       8133.700     8013.23             0.985189  
1      39635.000    36396.58             0.918294  
2    1042447.713   514840.56             0.493877  
3          2.000       13.46             6.730000  
4     402849.470    79580.14             0.197543  
5         13.000       16.80             1.292308  
6          2.000       30.12            15.060000  
7         16.000       13.50             0.843750  
8     415811.500   133785.38             0.321745  
9      62800.500    79712.18             1.269292  
10    123821.000   441976.19             3.569477  
11     23809.000    42794.71             1.797417

Mozeme si vsimnut ze najpouzivanejsie blokator vapnikovych kamenov s ucinnou latkou amlodipin je cesky Agen. Dalsimi ale menej pouzivanymi su slovinsky Amlodipin Stichting a nemecky Amlodipin-ratiopharm. Podla tabulky vidime, ze Slovensko taketo blokatory vapnikovych kamenov nevyraba.

# 7. Alprazolam

In [28]:

alprazolam = top20tab["N05BA12"].sort_values(by='Počet_balení', ascending=False)

for i,popis in enumerate(alprazolam['LIEK_popis']):
    if 'Alprazolam Orion' in popis:
        alprazolam.iloc[i,4] = 'Alprazolam Orion'
    elif 'FRONTIN' in popis:
        alprazolam.iloc[i,4] = 'FRONTIN'
    elif 'HELEX' in popis:
        alprazolam.iloc[i,4] = 'HELEX'
    elif 'Helex' in popis:
        alprazolam.iloc[i,4] = 'HELEX'
    elif 'XANAX' in popis:
        alprazolam.iloc[i,4] = 'XANAX'
    elif 'NEUROL SR' in popis:
        alprazolam.iloc[i,4] = 'NEUROL SR'
    elif 'NEUROL' in popis:
        alprazolam.iloc[i,4] = 'NEUROL'

alprazolam = alprazolam.groupby(['LIEK_popis', 'STAT_popis', 'VYROBCA_popis']).sum().reset_index()
display(alprazolam)

LIEK_popis STAT_popis                    VYROBCA_popis  index  \
0  Alprazolam Orion     Fínsko  Orion Corporation, Orion Pharma   7183   
1           FRONTIN   Maďarsko         EGIS Pharmaceuticals PLC  35945   
2             HELEX  Slovinsko            KRKA d.d., Novo mesto  43134   
3            NEUROL      Česko                    Zentiva, k.s.  21597   
4         NEUROL SR      Česko                    Zentiva a.s.    7196   
5             XANAX   Belgicko            Pfizer Europe MA EEIG  43202   

   Počet_balení  Úhrada_PAC  Cena úhrada_PAC/bal  
0          2.00        5.14             2.570000  
1     537957.20   731856.63             1.360437  
2      65145.00    65413.51             1.004122  
3    1056309.50  1395763.59             1.321359  
4          2.00        6.26             3.130000  
5     385706.03   647377.73             1.678423

In [76]:
fig = px.bar(alprazolam.query('Počet_balení > 50000'), x='LIEK_popis', y='Počet_balení', color='STAT_popis', title='Alprazolam nad 50 000')
fig.update_xaxes(tickangle=45)
fig.show()


 Najpuzivanejsim psycholeptikom s ucinnou latkou alprazolam je cesky Neurol. Priblizne o polovicu menej sa predavajuci madarsky Frontin a potom belgickz Xanax. Mozeme si vsimnut ze narozdiel od predchadzajuch grafov, ucinnu latku alprazolam pouziva menej liekov. Slovensko opat taketo psycholeptika nevyraba

# 8. Diklofenak

In [77]:
diklofenak = top20tab["M01AB05"].sort_values(by='Počet_balení', ascending=False)

for i,popis in enumerate(diklofenak['LIEK_popis']):
    if 'AKIS Plus' in popis:
        diklofenak.iloc[i,4] = 'AKIS Plus inj'
    elif 'AKIS' in popis:
        diklofenak.iloc[i,4] = 'AKIS inj'
    elif 'ALMIRAL inj' in popis:
        diklofenak.iloc[i,4] = 'ALMIRAL inj'
    elif 'ALMIRAL ' in popis:
        diklofenak.iloc[i,4] = 'ALMIRAL tbl'
    elif 'DICLAC RETARD tbl' in popis or 'Diclac retard' in popis:
        diklofenak.iloc[i,4] = 'DICLAC RETARD tbl'
    elif 'DICLOBENE' in popis and 'RETARD' in popis:
        diklofenak.iloc[i,4] = 'DICLOBENE RETARD tbl'
    elif 'DICLOBENE' in popis:
        diklofenak.iloc[i,4] = 'DICLOBENE tbl'
    elif 'DICLOFENAC AL RETARD' in popis:
        diklofenak.iloc[i,4] = 'DICLOFENAC AL RETARD tbl'
    elif 'DICLOFENAC AL' in popis:
        diklofenak.iloc[i,4] = 'DICLOFENAC AL tbl'
    elif 'DICLOFENAC DUO PHARMAVIT' in popis:
        diklofenak.iloc[i,4] = 'DICLOFENAC DUO PHARMAVIT cps'
    elif 'DICLOREUM tbl' in popis:
        diklofenak.iloc[i,4] = 'DICLOREUM tbl'
    elif 'DICLOREUM' in popis:
        diklofenak.iloc[i,4] = 'DICLOREUM sup'
    elif 'Dicuno' in popis:
        diklofenak.iloc[i,4] = 'Dicuno tbl'
    elif 'FLECTOR EP RAPID' in popis:
        diklofenak.iloc[i,4] = 'FLECTOR EP RAPID'
    elif 'VERAL' in popis and 'RETARD' in popis:
        diklofenak.iloc[i,4] = 'VERAL RETARD tbl'
    elif 'VERAL' in popis and 'tbl' in popis:
        diklofenak.iloc[i,4] = 'VERAL tbl'
    elif 'VERAL' in popis and 'sup' in popis:
        diklofenak.iloc[i,4] = 'VERAL sup'
    elif 'VERAL' in popis and 'sol' in popis:
        diklofenak.iloc[i,4] = 'VERAL sol inj'
    elif 'VERAL' in popis and 'gel' in popis:
        diklofenak.iloc[i,4] = 'VERAL gel'
    elif 'VOLTAREN 100 sup' in popis:
        diklofenak.iloc[i,4] = 'VOLTAREN sup'
    elif 'VOLTAREN 50 tbl' in popis:
        diklofenak.iloc[i,4] = 'VOLTAREN tbl'
    elif 'VOLTAREN RAPID' in popis:
        diklofenak.iloc[i,4] = 'VOLTAREN RAPID tbl'
    elif 'VOLTAREN' in popis and 'sol' in popis:
        diklofenak.iloc[i,4] = 'VOLTAREN sol ifc'
    elif 'Voltaren ActiGo Extra' in popis:
        diklofenak.iloc[i,4] = 'Voltaren ActiGo Extra tbl'
    elif 'Voltaren Actigo Extra' in popis:
        diklofenak.iloc[i,4] = 'Voltaren ActiGo Extra tbl'
    elif 'Voltaren Rapid' in popis:
        diklofenak.iloc[i,4] = 'Voltaren Rapid cps'
    elif 'VOLTAREN RETARD' in popis:
        diklofenak.iloc[i,4] = 'VOLTAREN RETARD tbl'
    
diklofenak = diklofenak.groupby(['LIEK_popis', 'STAT_popis', 'VYROBCA_popis']).sum().reset_index()
display(diklofenak)

LIEK_popis STAT_popis  \
0                  AKIS Plus inj  Slovensko   
1                       AKIS inj  Slovensko   
2                    ALMIRAL inj     Cyprus   
3                    ALMIRAL tbl     Cyprus   
4              DICLAC RETARD tbl  Slovinsko   
5           DICLOBENE RETARD tbl  Holandsko   
6                  DICLOBENE tbl  Holandsko   
7       DICLOFENAC AL RETARD tbl    Nemecko   
8              DICLOFENAC AL tbl    Nemecko   
9   DICLOFENAC DUO PHARMAVIT cps      Česko   
10                 DICLOREUM sup  Taliansko   
11                 DICLOREUM tbl  Taliansko   
12                    Dicuno tbl     Fínsko   
13              FLECTOR EP RAPID  Slovensko   
14              VERAL RETARD tbl      Česko   
15                     VERAL gel  Slovensko   
16                 VERAL sol inj  Slovensko   
17                     VERAL sup      Česko   
18                     VERAL tbl      Česko   
19            VOLTAREN RAPID tbl  Slovensko   
20           VOLTAREN RETARD tbl  Slovensko   
21              VOLTAREN sol ifc      Česko   
22                  VOLTAREN sup  Slovensko   
23                  VOLTAREN tbl  Slovensko   
24     Voltaren ActiGo Extra tbl      Česko   
25            Voltaren Rapid cps      Česko   

                                        VYROBCA_popis  index  Počet_balení  \
0                                IBSA Slovakia s.r.o.  21978      6095.903   
1                                IBSA Slovakia s.r.o.  21954       270.630   
2                                     Medochemie Ltd.  16503    167768.830   
3                                     Medochemie Ltd.   5499     28693.990   
4                         Sandoz Pharmaceuticals d.d.  10966     11883.500   
5                                           TEV-NL-4?  16407     57957.000   
6                                           TEV-NL-4?  21862     46020.650   
7                                   ALIUD PHARMA GmbH  11011        23.000   
8                          Aliud Pharma GmbH & Co. KG  11013         4.000   
9                  PharmaSwiss Česká republika s.r.o.   5510    170025.010   
10                             Alfa Wassermann S.p.A.   5513         3.000   
11                             Alfa Wassermann S.p.A.   5512        12.000   
12                                      Vitabalans Oy  10923      2504.797   
13                               IBSA Slovakia s.r.o.  10983    825870.470   
14                          Herbacos Recordati s.r.o.  10965     80757.530   
15                                       Zentiva a.s.  16440       368.000   
16                                  Slovakofarma a.s.   5481         6.000   
17                          Herbacos Recordati s.r.o.   5474         1.000   
18                          Herbacos Recordati s.r.o.  27416     64051.168   
19                           Novartis Slovakia s.r.o.   5480    163277.900   
20                           Novartis Slovakia s.r.o.   5507      9310.960   
21                                    Novartis s.r.o.   5508       102.500   
22                           Novartis Slovakia s.r.o.   5498      5866.000   
23                           Novartis Slovakia s.r.o.   5511      8390.000   
24  GlaxoSmithKline Consumer Healthcare Czech Repu...  10983    126203.500   
25  GlaxoSmithKline Consumer Healthcare Czech Repu...  16428    238259.180   

    Úhrada_PAC  
0     51936.89  
1      3414.97  
2    104943.58  
3     23225.76  
4     15556.74  
5     29930.63  
6     25987.73  
7       104.92  
8        14.11  
9    469116.41  
10        6.30  
11       32.90  
12     9412.28  
13  2762210.14  
14   196583.81  
15     2012.08  
16        0.00  
17        4.10  
18   183991.73  
19   145603.20  
20    54202.95  
21      203.07  
22     6651.44  
23    23238.67  
24   668509.79  
25  1285498.49

In [79]:
fig = px.bar(diklofenak.query('Počet_balení > 50000'), x='LIEK_popis', y='Počet_balení', color='STAT_popis', title='Diklofenak nad 50 000')
fig.update_xaxes(tickangle=45)
fig.show()

Z grafu vidime, ze najpredavanejsi je slovensky Flector Ep rapid. V tejto oblasti su bohato zastupene slovenske a ceske lieky.

In [81]:
fig = px.pie(diklofenak.groupby('STAT_popis').sum().reset_index(), names='STAT_popis', values='Počet_balení', title = 'Diklofenak podla krajiny vyrobcu (vsetky)')
fig.show()

# 9. Metamizol, sodna sol

In [82]:

metamizol = top20tab["N02BB02"].sort_values(by='Počet_balení', ascending=False)

for i,popis in enumerate(metamizol['LIEK_popis']):
    if 'METAMISTAD' in popis:
        metamizol.iloc[i,4] = 'METAMISTAD tbl'
    elif 'Metamilan' in popis:
        metamizol.iloc[i,4] = 'Metamilan tbl'
    elif 'Metamizol Stada' in popis:
        metamizol.iloc[i,4] = 'Metamizol Stada'
    elif 'Metamizol Teva' in popis:
        metamizol.iloc[i,4] = 'Metamizol Teva'
    elif 'Metamizole Kalceks' in popis:
        metamizol.iloc[i,4] = 'Metamizole Kalceks'
    elif 'NOVALGIN' in popis and 'tbl' in popis:
        metamizol.iloc[i,4] = 'NOVALGIN tbl'
    elif 'NOVALGIN' in popis:
        metamizol.iloc[i,4] = 'NOVALGIN sol inj'
    

metamizol = metamizol.groupby(['LIEK_popis', 'STAT_popis', 'VYROBCA_popis']).sum().reset_index()

display(metamizol.sort_values(by='LIEK_popis'))

LIEK_popis STAT_popis                    VYROBCA_popis  index  \
0      METAMISTAD tbl    Nemecko            STADA Arzneimittel AG  12509   
1       Metamilan tbl    Rakúsko                  G.L.Pharma GmbH   6249   
2     Metamizol Stada    Nemecko            STADA Arzneimittel AG  12513   
3      Metamizol Teva  Holandsko                        TEV-NL-4?   6244   
4  Metamizole Kalceks   Lotyšsko                       AS KALCEKS  12501   
5    NOVALGIN sol inj    Nemecko  Aventis Pharma Deutschland GmbH   6253   
6    NOVALGIN sol inj  Slovensko   sanofi-aventis Slovakia s.r.o.  12491   
7        NOVALGIN tbl  Slovensko   sanofi-aventis Slovakia s.r.o.  12495   
8        NOVALGIN tbl  Taliansko            Aventis Pharma S.p.A.   6252   

   Počet_balení  Úhrada_PAC  
0    183829.997    74327.59  
1     32683.300     8515.62  
2    112570.900    96246.06  
3         1.000        0.10  
4     56516.500     1313.88  
5         3.000        0.00  
6    218420.700   162095.26  
7   1342223.310  1574153.23  
8         2.000        4.24

In [84]:
fig = px.bar(metamizol.query('Počet_balení > 100000'), x='LIEK_popis', y='Počet_balení', color='STAT_popis', title='Metamizol, sodna sol nad 100 000')
fig.update_xaxes(tickangle=45)
fig.show()

Z grafu vidime, ze sa najviac predava slovensky Novalgin v tabletach. 

# 10. Metoprolol

In [99]:
metoprolol = top20tab["C07AB02"].sort_values(by='Počet_balení', ascending=False)

for i,popis in enumerate(metoprolol['LIEK_popis']):
    if 'BETALOC' in popis:
        metoprolol.iloc[i,4] = 'BETALOC'
    elif 'CORVITOL' in popis:
        metoprolol.iloc[i,4] = 'CORVITOL'
    elif 'EGILOK' in popis:
        metoprolol.iloc[i,4] = 'EGILOK'
    elif 'Egilok' in popis:
        metoprolol.iloc[i,4] = 'EGILOK'
    elif 'Metoprobene' in popis:
        metoprolol.iloc[i,4] = 'METOPROBENE'
    elif 'Metoprolol Polpharma' in popis:
        metoprolol.iloc[i,4] = 'Metoprolol Polpharma'
    elif 'Metoprolol ratiopharm' in popis:
        metoprolol.iloc[i,4] = 'Metoprolol ratiopharm'
    elif 'VASOCARDIN' in popis:
        metoprolol.iloc[i,4] = 'VASOCARDIN'

metoprolol = metoprolol.groupby(['LIEK_popis', 'STAT_popis']).sum().reset_index()
display(metoprolol.sort_values(by='LIEK_popis'))

LIEK_popis STAT_popis  index  Počet_balení  Úhrada_PAC
0                BETALOC  Taliansko   7783    917306.500  2991813.86
1               CORVITOL    Nemecko   1958     97308.000    13077.12
2                 EGILOK   Maďarsko  21348    157033.400   136782.42
3            METOPROBENE    Nemecko   1947         1.000        6.72
4   Metoprolol Polpharma     Poľsko   1948         3.000       23.22
5  Metoprolol ratiopharm    Nemecko   7818    368367.099   134140.37
6             VASOCARDIN  Slovensko   7781    333583.520    27500.64

In [101]:
fig = px.bar(metoprolol.query('Počet_balení > 50000'), x='LIEK_popis', y='Počet_balení', color='STAT_popis', title='Metoprolol nad 50 000')
fig.update_xaxes(tickangle=45)
fig.show()

Preco nie je slovensky najpredavanejsi?

# 11. Kyselina acetylsalicylova

In [85]:

k_acetylsalicylova = top20tab["B01AC06"].sort_values(by='Počet_balení', ascending=False)

for i,popis in enumerate(k_acetylsalicylova['LIEK_popis']):
    if 'ANOPYRIN' in popis:
        k_acetylsalicylova.iloc[i,4] = 'ANOPYRIN'
    elif 'ASPIRIN PROTECT' in popis:
        k_acetylsalicylova.iloc[i,4] = 'ASPIRIN PROTECT'
    elif 'CARDIORAL' in popis:
        k_acetylsalicylova.iloc[i,4] = 'CARDIORAL'
    elif 'Gerasa' in popis:
        k_acetylsalicylova.iloc[i,4] = 'Gerasa'
    elif 'Kyselina acetylsalicylová Xantis' in popis:
        k_acetylsalicylova.iloc[i,4] = 'Kyselina acetylsalicylová Xantis'
    elif 'Preventax' in popis:
        k_acetylsalicylova.iloc[i,4] = 'Preventax'
    elif 'STADAPYRIN' in popis:
        k_acetylsalicylova.iloc[i,4] = 'STADAPYRIN'
    elif 'Vasopirin' in popis:
        k_acetylsalicylova.iloc[i,4] = 'Vasopirin'
   

k_acetylsalicylova = k_acetylsalicylova.groupby(['LIEK_popis', 'STAT_popis', 'VYROBCA_popis']).sum().reset_index()

display(k_acetylsalicylova.sort_values(by='LIEK_popis'))

LIEK_popis STAT_popis  \
0                          ANOPYRIN  Slovensko   
1                   ASPIRIN PROTECT  Slovensko   
2                         CARDIORAL  Slovensko   
3                            Gerasa    Rakúsko   
4  Kyselina acetylsalicylová Xantis     Cyprus   
5                         Preventax  Holandsko   
6                        STADAPYRIN    Nemecko   
7                         Vasopirin      Česko   

                        VYROBCA_popis  index  Počet_balení  Úhrada_PAC  
0                           ZNT-SK-2?   6040   1089219.322  2058022.70  
1                 Bayer, spol. s.r.o.   3986    329049.400   877523.76  
2                IBSA Slovakia s.r.o.   1015         4.000      186.87  
3                     G.L.Pharma GmbH   2027      3931.160     6697.88  
4               XANTIS PHARMA LIMITED   2019     10972.540    11696.24  
5                           TEV-NL-4?   1016     76991.310   215135.04  
6               STADA Arzneimittel AG   2010    201492.075   554253.19  
7  PharmaSwiss Česká republika s.r.o.   5025     39774.200    79233.23

In [87]:
fig = px.bar(k_acetylsalicylova.query('Počet_balení > 50000'), x='LIEK_popis', y='Počet_balení', color='STAT_popis', title='Kyselina acetylsalicylova nad 50 000')
fig.update_xaxes(tickangle=45)
fig.show()

In [ ]:
Najpredavanejsim liekom s kyselinou acetylsalicylovou je slovensky Anopyrin. Najviac sa predavaju slovenske lieky, ktorych je na trhu najviac.

# 12. Atorvastatin (top 55)

In [88]:
atorvastatin = top20tab["C10AA05"].sort_values(by='Počet_balení', ascending=False).head(55)

for i,popis in enumerate(atorvastatin['LIEK_popis']):
    if 'ATORIS' in popis:
        atorvastatin.iloc[i,4] = 'ATORIS'
    if 'Atoris' in popis:
        atorvastatin.iloc[i,4] = 'ATORIS'
    if 'Atorvastatin Actavis' in popis:
        atorvastatin.iloc[i,4] = 'Atorvastatin Actavis'
    if 'SORTIS' in popis:
        atorvastatin.iloc[i,4] = 'SORTIS'
    if 'Torvacard Novum' in popis:
        atorvastatin.iloc[i,4] = 'Torvacard Novum'
    if 'Tulip' in popis:
        atorvastatin.iloc[i,4] = 'Tulip'
    if 'Atorvastatin Mylan' in popis:
        atorvastatin.iloc[i,4] = 'Atorvastatin Mylan'
    if 'Atorvastatin STADA' in popis:
        atorvastatin.iloc[i,4] = 'Atorvastatin STADA'
    if 'Atorvastatin Teva Pharma' in popis:
        atorvastatin.iloc[i,4] = 'Atorvastatin Teva Pharma'
    if 'Atorvastatin Saneca' in popis:
        atorvastatin.iloc[i,4] = 'Atorvastatin Saneca'
    if 'Gletor' in popis:
        atorvastatin.iloc[i,4] = 'Gletor'

atorvastatin = atorvastatin.groupby(['LIEK_popis', 'STAT_popis', 'VYROBCA_popis']).sum().reset_index()

display(atorvastatin.sort_values(by='LIEK_popis'))

LIEK_popis STAT_popis                         VYROBCA_popis  \
0                    ATORIS  Slovinsko                 KRKA d.d., Novo mesto   
1      Atorvastatin Actavis     Island                 Actavis Group PTC ehf   
2        Atorvastatin Mylan      ĺrsko                 Mylan Ireland Limited   
3        Atorvastatin STADA    Nemecko                 STADA Arzneimittel AG   
4       Atorvastatin Saneca     Cyprus                 XANTIS PHARMA LIMITED   
5  Atorvastatin Teva Pharma  Slovensko  TEVA Pharmaceuticals Slovakia s.r.o.   
6                    Gletor      Česko       Glenmark Pharmaceuticals s.r.o.   
7                    SORTIS   Belgicko                 Pfizer Europe MA EEIG   
8           Torvacard Novum      Česko                         Zentiva, k.s.   
9                     Tulip  Slovinsko           Sandoz Pharmaceuticals d.d.   

   index  Počet_balení  Úhrada_PAC  
0  31395    561480.081  1159087.78  
1  20047    282689.050   229382.68  
2   8457     18970.680    25666.59  
3  11574     25251.000    46411.62  
4  11646      7698.654     1390.62  
5  14197     24108.160     9229.01  
6   5773      4298.520     7921.31  
7   8631    128401.967   180299.01  
8  22781    363483.663   639138.51  
9  22706    152051.083   307103.12

In [90]:
fig = px.bar(atorvastatin.query('Počet_balení > 100000'), x='LIEK_popis', y='Počet_balení', color='STAT_popis', title='Atorovastatin nad 100 000')
fig.update_xaxes(tickangle=45)
fig.show()

Najpredavanejsi je slovinsky Atoris, avsak tiez celkom predavanym je aj cesky Torvacard Novum

# 13. Zolpidem

In [105]:

zolpidem = top20tab["N05CF02"].sort_values(by='Počet_balení', ascending=False).head(55)

for i,popis in enumerate(zolpidem['LIEK_popis']):
    if 'ADORMA' in popis:
        zolpidem.iloc[i,4] = 'ADORMA'
    elif 'HYPNOGEN' in popis:
        zolpidem.iloc[i,4] = 'HYPNOGEN'
    elif 'SANVAL' in popis:
        zolpidem.iloc[i,4] = 'SANVAL'
    elif 'STILNOX' in popis:
        zolpidem.iloc[i,4] = 'STILNOX'
    elif 'ZOLPIDEM MERCK' in popis:
        zolpidem.iloc[i,4] = 'ZOLPIDEM MERCK'
    elif 'Zolpidem Orion' in popis:
        zolpidem.iloc[i,4] = 'Zolpidem Orion'
    elif 'Zolpidem Pfizer' in popis:
        zolpidem.iloc[i,4] = 'Zolpidem Pfizer'
    elif 'Zolpidem Saneca' in popis:
        zolpidem.iloc[i,4] = 'Zolpidem Saneca'
    elif 'Zolpidem Teva' in popis:
        zolpidem.iloc[i,4] = 'Zolpidem Teva'
    elif 'Zolpidem Vitabalans' in popis:
        zolpidem.iloc[i,4] = 'Zolpidem Vitabalans'
    elif 'Zolpinox' in popis:
        zolpidem.iloc[i,4] = 'Zolpinox'

zolpidem = zolpidem.groupby(['LIEK_popis', 'STAT_popis', 'VYROBCA_popis']).sum().reset_index()

display(zolpidem.sort_values(by='LIEK_popis'))

LIEK_popis STAT_popis                     VYROBCA_popis  index  \
0                ADORMA  Slovinsko  Alkaloid - INT d.o.o., Ljubljana   7246   
1              HYPNOGEN    Nemecko                       Takeda GmbH  14504   
2                SANVAL  Slovinsko       Sandoz Pharmaceuticals d.d.  14489   
3               STILNOX  Slovensko    sanofi-aventis Slovakia s.r.o.  21756   
4        ZOLPIDEM MERCK      ĺrsko             Mylan Ireland Limited   7256   
5        Zolpidem Orion     Fínsko                 Orion Corporation  14497   
6       Zolpidem Pfizer  Taliansko              PFIZER Italia S.r.l.   7240   
7       Zolpidem Saneca     Cyprus             XANTIS PHARMA LIMITED  14505   
8         Zolpidem Teva  Holandsko                        Synthon BV   7254   
9   Zolpidem Vitabalans     Fínsko                     Vitabalans Oy  21726   
10             Zolpinox      Česko        Neuraxpharm Bohemia s.r.o.  28950   

    Počet_balení  Úhrada_PAC  
0          1.000        2.00  
1     249315.840   407727.12  
2     242727.100   231796.13  
3     866042.660  1450471.81  
4      97213.240   154214.22  
5          9.000       12.45  
6          3.000        7.65  
7      20968.000    18911.62  
8          1.500        3.65  
9       3381.300     6249.51  
10     75791.287    85728.27

In [107]:
fig = px.bar(zolpidem.query('Počet_balení > 50000'), x='LIEK_popis', y='Počet_balení', color='STAT_popis', title='Zolpidem nad 50 000')
fig.update_xaxes(tickangle=45)
fig.show()

# 14. Xylometazolín

In [113]:
xylometazolin = top20tab["R01AA07"].sort_values(by='Počet_balení', ascending=False)

for i,popis in enumerate(xylometazolin['LIEK_popis']):
    if 'DR.RENTSCHLER SCHNUPFENSPRAY' in popis:
        xylometazolin.iloc[i,4] = 'DR.RENTSCHLER SCHNUPFENSPRAY'
    elif 'Maresyl' in popis:
        xylometazolin.iloc[i,4] = 'Maresyl'
    elif 'NASENSPRAY' in popis:
        xylometazolin.iloc[i,4] = 'NASENSPRAY'
    elif 'NASENTROPFEN' in popis:
        xylometazolin.iloc[i,4] = 'NASENTROPFEN'
    elif 'OLYNTH HA' in popis:
        xylometazolin.iloc[i,4] = 'OLYNTH HA'
    elif 'OLYNTH' in popis:
        xylometazolin.iloc[i,4] = 'OLYNTH'
    elif 'OTRIVIN' in popis:
        xylometazolin.iloc[i,4] = 'OTRIVIN'
    elif 'Orinox' in popis:
        xylometazolin.iloc[i,4] = 'Orinox'
    elif 'Otrivin Menthol' in popis:
        xylometazolin.iloc[i,4] = 'Otrivin Menthol'
    elif 'Rinoxyl' in popis:
        xylometazolin.iloc[i,4] = 'Rinoxyl'
    elif 'mar rhino' in popis:
        xylometazolin.iloc[i,4] = 'mar rhino'    

xylometazolin = xylometazolin.groupby(['LIEK_popis', 'STAT_popis', 'VYROBCA_popis']).sum().reset_index()

display(xylometazolin.sort_values(by='LIEK_popis'))

LIEK_popis          STAT_popis  \
0   DR.RENTSCHLER SCHNUPFENSPRAY             Nemecko   
1                        Maresyl          Francúzsko   
2                     NASENSPRAY             Nemecko   
3                     NASENSPRAY             Nemecko   
4                   NASENTROPFEN             Nemecko   
5                         OLYNTH          Francúzsko   
6                         OLYNTH  Spojené kráľovstvo   
7                         OLYNTH               ĺrsko   
8                      OLYNTH HA               ĺrsko   
9                        OTRIVIN               Česko   
10                        Orinox               Česko   
11               Otrivin Menthol               Česko   
12                       Rinoxyl              Fínsko   
13                     mar rhino             Nemecko   

                                        VYROBCA_popis  index  Počet_balení  \
0              Dr. Rentschler Arzneimittel GmbH & Co.   7797         127.0   
1                              Mylan Laboratories SAS  15583       10395.0   
2                          Aliud Pharma GmbH & Co. KG   7811           4.0   
3                               STADA Arzneimittel AG   7813           1.0   
4                          Aliud Pharma GmbH & Co. KG   7805           2.0   
5                                          PFIZER PGM  15605          86.0   
6   McNeil Products Limited C/O Johnson & Johnson ...   7801           1.0   
7                 McNeil Healthcare (Ireland) Limited  15618      621781.0   
8                 McNeil Healthcare (Ireland) Limited  15599      529046.0   
9   GlaxoSmithKline Consumer Healthcare Czech Repu...  31178       95259.5   
10                              Dr. Max Pharma s.r.o.   7804           1.0   
11  GlaxoSmithKline Consumer Healthcare Czech Repu...   7798      133135.0   
12                    Orion Corporation, Orion Pharma  15619           3.0   
13                              STADA Arzneimittel AG  15615      106986.0   

    Úhrada_PAC  
0       909.99  
1     35427.43  
2        12.95  
3         8.30  
4         9.45  
5       349.84  
6         0.06  
7   2336426.43  
8   2541167.62  
9    450124.85  
10        3.00  
11   681134.20  
12       40.08  
13   401803.74

In [115]:
fig = px.bar(xylometazolin.query('Počet_balení > 50000'), x='LIEK_popis', y='Počet_balení', color='STAT_popis', title='Xylometazolin nad 50 000')
fig.update_xaxes(tickangle=45)
fig.show()

# 15. Rôzne iné krčné antiseptiká

In [124]:
ine_krcne = top20tab["R02AA20"].sort_values(by='Počet_balení', ascending=False)

for i,popis in enumerate(ine_krcne['LIEK_popis']):
    if 'Dorithricin pas' in popis:
        ine_krcne.iloc[i,4] = 'Dorithricin pas'
    elif 'Faryngal 5 mg/1 mg lisované' in popis:
        ine_krcne.iloc[i,4] = 'Faryngal pas'
    elif 'Faryngal' in popis:
        ine_krcne.iloc[i,4] = 'Faryngal aerodis'
    elif 'HEXORALETTEN' in popis:
        ine_krcne.iloc[i,4] = 'HEXORALETTEN N pas'
    elif 'NEO-ANGIN' in popis:
        ine_krcne.iloc[i,4] = 'NEO-ANGIN pas'
    elif 'SEPTISAN tbl' in popis:
        ine_krcne.iloc[i,4] = 'SEPTISAN tbl'
    elif 'SEPTOLETE' in popis:
        ine_krcne.iloc[i,4] = 'SEPTOLETE'
    elif 'STOPANGIN' in popis:
        ine_krcne.iloc[i,4] = 'STOPANGIN'
    elif 'STREPSILS' in popis:
        ine_krcne.iloc[i,4] = 'STREPSILS'
    elif 'Septolete' in popis:
        ine_krcne.iloc[i,4] = 'SEPTOLETE'
    elif 'Strepsils' in popis:
        ine_krcne.iloc[i,4] = 'SEPTOLETE'
    elif 'neo-angin' in popis:
        ine_krcne.iloc[i,4] = 'NEO-ANGIN'
      

ine_krcne = ine_krcne.groupby(['LIEK_popis', 'STAT_popis', 'VYROBCA_popis']).sum().reset_index()

display(ine_krcne.sort_values(by='LIEK_popis'))

LIEK_popis          STAT_popis  \
0      Dorithricin pas             Nemecko   
1     Faryngal aerodis           Slovinsko   
2         Faryngal pas           Slovinsko   
3   HEXORALETTEN N pas               ĺrsko   
4            NEO-ANGIN             Nemecko   
5        NEO-ANGIN pas             Nemecko   
6         SEPTISAN tbl           Slovensko   
7            SEPTOLETE           Slovinsko   
8            SEPTOLETE               Česko   
9            STOPANGIN               Česko   
10           STREPSILS  Spojené kráľovstvo   
11           STREPSILS               Česko   

                                        VYROBCA_popis   index  Počet_balení  \
0                         Medice Pharma GmbH & Co. KG   15809    653457.650   
1                         Sandoz Pharmaceuticals d.d.    7929      3846.000   
2                         Sandoz Pharmaceuticals d.d.   15795     11373.370   
3                 McNeil Healthcare (Ireland) Limited    7901      3768.500   
4                                     Divapharma GmbH    7907     11357.380   
5                                     Divapharma GmbH   31692    105924.680   
6                      sanofi-aventis Slovakia s.r.o.    7936     90881.000   
7                               KRKA d.d., Novo mesto   63315    109671.600   
8   Reckitt Benckiser (Czech Republic), spol. s r.o.     7919     13398.600   
9                        Teva Czech Industries s.r.o.   15802     63500.500   
10                     Boots Healthcare International   31636       143.000   
11  Reckitt Benckiser (Czech Republic), spol. s r.o.   118810    376971.679   

    Úhrada_PAC  
0   4647914.33  
1     14446.95  
2     33328.57  
3      5789.35  
4     68407.89  
5    680358.98  
6    301647.58  
7    601482.25  
8     88941.37  
9    323868.66  
10      880.26  
11  2481959.72

In [126]:
fig = px.bar(ine_krcne.query('Počet_balení > 50000'), x='LIEK_popis', y='Počet_balení', color='STAT_popis', title='Rôzne iné krčné antiseptiká nad 50 000')
fig.update_xaxes(tickangle=45)
fig.show()

In [135]:
strepsils = top20tab["R02AA20"].sort_values(by='Počet_balení', ascending=False)

for i,popis in enumerate(strepsils['LIEK_popis']):
    if 'STREPSILS' not in popis and 'Strepsils' not in popis:
        strepsils.iloc[i,4] = 'ine'
      
strepsils = strepsils.sort_values(by='LIEK_popis').iloc[0:20,:]

for i,popis in enumerate(strepsils['LIEK_popis']):
    if 'CITRON A BYLINY' in popis:
        strepsils.iloc[i,4] = 'CITRON A BYLINY'
    elif 'COOLING MINT' in popis:
        strepsils.iloc[i,4] = 'COOLING MINT'
    elif 'MED A CITRÓN' in popis:
        strepsils.iloc[i,4] = 'MED A CITRÓN'
    elif 'Med a Citrón' in popis:
        strepsils.iloc[i,4] = 'MED A CITRÓN'
    elif 'Menthol a Eucalyptus' in popis:
        strepsils.iloc[i,4] = 'Menthol a Eucalyptus'
    elif 'ORIGINAL' in popis:
        strepsils.iloc[i,4] = 'ORIGINAL'
    elif 'PLUS' in popis:
        strepsils.iloc[i,4] = 'PLUS'
    elif 'POMARANČ S VITAMÍNOM C' in popis:
        strepsils.iloc[i,4] = 'POMARANČ S VITAMÍNOM C'
    elif 'Vitamin C' in popis:
        strepsils.iloc[i,4] = 'Vitamin C'
    elif 'Jahoda bez cukru' in popis:
        strepsils.iloc[i,4] = 'Jahoda bez cukru'
strepsils = strepsils.groupby(['LIEK_popis']).sum().reset_index()

display(strepsils)

LIEK_popis  index  Počet_balení  Úhrada_PAC
0         CITRON A BYLINY  15845     44667.250   301320.29
1            COOLING MINT   7935     44157.520   280492.45
2        Jahoda bez cukru   7919     13398.600    88941.37
3            MED A CITRÓN  31677    149374.240   945308.40
4    Menthol a Eucalyptus  39560      2433.500    14849.97
5                ORIGINAL  15845       110.500      691.05
6                    PLUS  15821     76760.689   566793.42
7  POMARANČ S VITAMÍNOM C   7934        26.500      163.60
8               Vitamin C  15829     59584.480   373220.80

In [139]:
fig = px.bar(strepsils, x='LIEK_popis', y='Počet_balení', title='Ktora prichut Strepsils pastiliek je najpredavanejsia?')
fig.update_xaxes(tickangle=45)
fig.show()

In [140]:
fig = px.pie(strepsils, names='LIEK_popis', values='Počet_balení', title='Ktora prichut Strepsils pastiliek je najpredavanejsia?')
fig.show()